fix: Missing slots attributes, wrong variable type, duplicate http methods (#2500)
Pycord-Development/pycord: CHANGELOG.md

These changes are available on the `master` branch, but have not yet been released.

### Fixed

> - Fixed missed `stacklevel` parameter in `warn_deprecated` function call inside
>   `deprecated` decorator in utils.

Reviewer: ```suggestion
- Fixed missing `stacklevel` parameter in `warn_deprecated` function call inside
  `deprecated` decorator in utils.
```

  ([#2500](https://github.com/Pycord-Development/pycord/pull/2500))

> - Fixed typehint in `ConnectionState._polls` incorrectly specified as `Guild` instead of
>   `Poll`. ([#2500](https://github.com/Pycord-Development/pycord/pull/2500))

Reviewer: ```suggestion
- Fixed the typehint in `ConnectionState._polls` to reflect actual behavior, changing it from `Guild` to `Poll`. ([#2500](https://github.com/Pycord-Development/pycord/pull/2500))
```

- Fixed missing `__slots__` attributes in `RawReactionClearEmojiEvent` and
  `RawMessagePollVoteEvent` classes.
  ([#2500](https://github.com/Pycord-Development/pycord/pull/2500))

> - Fixed `default_sort_order` variable type being `int` instead of typehinted `SortOrder`

Reviewer: ```suggestion
- Fixed `default_sort_order` attribute type being `int` instead of typehinted `SortOrder`
```
